File: examples/delphi/ViewMain.pas
(***** BEGIN LICENSE BLOCK *****
 * Version: MPL 1.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * The Original Code is TurboPower Async Professional
 *
 * The Initial Developer of the Original Code is
 * TurboPower Software
 *
 * Portions created by the Initial Developer are Copyright (C) 1991-2002
 * the Initial Developer. All Rights Reserved.
 *
 * Contributor(s):
 *
 * ***** END LICENSE BLOCK ***** *)

{**********************************************************}
{*                     VIEWMAIN.PAS                       *}
{**********************************************************}

{**********************Description************************}
{* A fax viewer that allows you to view APF files.       *}
{*********************************************************}

unit ViewMain;

interface

uses
  WinTypes,
  WinProcs,
  Messages,
  SysUtils,
  Classes,
  Graphics,
  Controls,
  Forms,
  Dialogs,
  Menus,
  ExtCtrls,
  Buttons,
  StdCtrls,
  ooMisc,
  AdFView,
  AdFaxPrn,
  AdFPStat,
  WComp,
  Percent;

type
  TMainForm = class(TForm)
    MainMenu: TMainMenu;
    FaxViewer: TApdFaxViewer;
    FileItem: TMenuItem;
    OpenItem: TMenuItem;
    N1: TMenuItem;
    PrintSetupItem: TMenuItem;
    PrintItem: TMenuItem;
    N2: TMenuItem;
    ExitItem: TMenuItem;
    EditItem: TMenuItem;
    SelectAllItem: TMenuItem;
    CopyItem: TMenuItem;
    ViewItem: TMenuItem;
    N25PercentItem: TMenuItem;
    N50PercentItem: TMenuItem;
    N75PercentItem: TMenuItem;
    N100PercentItem: TMenuItem;
    N200PercentItem: TMenuItem;
    N400PercentItem: TMenuItem;
    N3: TMenuItem;
    OtherSizeItem: TMenuItem;
    WhitespaceCompOption: TMenuItem;
    ZoomInItem: TMenuItem;
    ZoomOutItem: TMenuItem;
    N4: TMenuItem;
    OpenDialog: TOpenDialog;
    FaxPrinter: TApdFaxPrinter;
    PrinterStatus: TApdFaxPrinterStatus;
    N5: TMenuItem;
    Rotate90Item: TMenuItem;
    Rotate180Item: TMenuItem;
    Rotate270Item: TMenuItem;
    NoRotateItem: TMenuItem;
    StatusPanel: TPanel;
    CloseItem: TMenuItem;
    Panel1: TPanel;
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    CheckBox3: TCheckBox;
    SpeedButton1: TSpeedButton;
    SpeedButton2: TSpeedButton;
    procedure OpenItemClick(Sender: TObject);
    procedure PrintSetupItemClick(Sender: TObject);
    procedure PrintItemClick(Sender: TObject);
    procedure ExitItemClick(Sender: TObject);
    procedure SelectAllItemClick(Sender: TObject);
    procedure CopyItemClick(Sender: TObject);
    procedure ZoomInItemClick(Sender: TObject);
    procedure ZoomOutItemClick(Sender: TObject);
    procedure N25PercentItemClick(Sender: TObject);
    procedure N50PercentItemClick(Sender: TObject);
    procedure N75PercentItemClick(Sender: TObject);
    procedure N100PercentItemClick(Sender: TObject);
    procedure N200PercentItemClick(Sender: TObject);
    procedure N400PercentItemClick(Sender: TObject);
    procedure OtherSizeItemClick(Sender: TObject);
    procedure WhitespaceCompOptionClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure NoRotateItemClick(Sender: TObject);
    procedure Rotate90ItemClick(Sender: TObject);
    procedure Rotate180ItemClick(Sender: TObject);
    procedure Rotate270ItemClick(Sender: TObject);
    procedure FaxViewerPageChange(Sender: TObject);
    procedure FaxViewerViewerError(Sender: TObject; ErrorCode: Integer);
    procedure CloseItemClick(Sender: TObject);
    procedure NextClick(Sender: TObject);
    procedure PrevClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
    procedure FaxViewerDropFile(Sender: TObject; FileName: String);
  private
    { Private declarations }
  public
    ViewPercent : Integer;

    procedure UpdateViewPercent(const NewPercent : Integer);
    procedure EnableZoomChoices;
    procedure DisableZoomChoices;
    procedure UncheckZoomChoices;
    procedure EnableRotationChoices;
    procedure DisableRotationChoices;
    procedure UncheckRotationChoices;
    procedure OpenFile(const FileName : string);
    procedure CloseFile;
  end;

var
  MainForm: TMainForm;

implementation

{$R *.DFM}

procedure TMainForm.CloseFile;
begin
  FaxViewer.FileName := '';
  DisableZoomChoices;
  DisableRotationChoices;
  SelectAllItem.Enabled := False;
  CopyItem.Enabled      := False;
  StatusPanel.Caption   := '  No file loaded';
  FaxViewer.Invalidate;
end;

procedure TMainForm.OpenFile(const FileName : string);
begin
  FaxViewer.BeginUpdate;
  FaxViewer.Scaling   := False;
  FaxViewer.HorizMult := 1;
  FaxViewer.HorizDiv  := 1;
  FaxViewer.VertMult  := 1;
  FaxViewer.VertDiv   := 1;
  FaxViewer.EndUpdate;
  UncheckZoomChoices;
  UncheckRotationChoices;
  try
    FaxViewer.FileName := FileName;
    EnableZoomChoices;
    EnableRotationChoices;
    SelectAllItem.Enabled   := True;
    CopyItem.Enabled        := True;
    N100PercentItem.Checked := True;
    NoRotateItem.Checked    := True;
    ViewPercent             := 100;
    StatusPanel.Caption     := Format('  Viewing page 1 of %d in %s', [FaxViewer.NumPages, FaxViewer.FileName]);
  except
    CloseFile;
    MessageDlg('Error opening fax file '+FileName, mtError, [mbOK], 0);
  end;
end;

procedure TMainForm.OpenItemClick(Sender: TObject);
begin
  if OpenDialog.Execute then
    OpenFile(OpenDialog.FileName);
end;

procedure TMainForm.CloseItemClick(Sender: TObject);
begin
  CloseFile;
end;

procedure TMainForm.PrintSetupItemClick(Sender: TObject);
begin
  FaxPrinter.PrintSetup;
end;

procedure TMainForm.PrintItemClick(Sender: TObject);
begin
  if (FaxViewer.FileName <> '') then begin
    FaxPrinter.FileName := FaxViewer.FileName;
    FaxPrinter.PrintFax;
  end;
end;

procedure TMainForm.ExitItemClick(Sender: TObject);
begin
  Close;
end;

procedure TMainForm.SelectAllItemClick(Sender: TObject);
begin
  FaxViewer.SelectImage;
end;

procedure TMainForm.CopyItemClick(Sender: TObject);
begin
  FaxViewer.CopyToClipBoard;
end;

procedure TMainForm.ZoomInItemClick(Sender: TObject);
var
  TempPercent : Integer;

begin
  if ((ViewPercent mod 25) = 0) then
    TempPercent := ViewPercent + 25
  else
    TempPercent := ViewPercent + (25 - (ViewPercent mod 25));
  if (TempPercent > 400) then begin
    MessageBeep(0);
    Exit;
  end;
  UpdateViewPercent(TempPercent);
end;

procedure TMainForm.ZoomOutItemClick(Sender: TObject);
var
  TempPercent : Integer;

begin
  if ((ViewPercent mod 25) = 0) then
    TempPercent := ViewPercent - 25
  else
    TempPercent := ViewPercent - (25 - (ViewPercent mod 25));

  if (TempPercent < 25) then begin
    MessageBeep(0);
    Exit;
  end;
  UpdateViewPercent(TempPercent);
end;

procedure TMainForm.N25PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(25);
end;

procedure TMainForm.N50PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(50);
end;

procedure TMainForm.N75PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(75);
end;

procedure TMainForm.N100PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(100);
end;

procedure TMainForm.N200PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(200);
end;

procedure TMainForm.N400PercentItemClick(Sender: TObject);
begin
  UpdateViewPercent(400);
end;

procedure TMainForm.OtherSizeItemClick(Sender: TObject);
var
  Frm         : TPercentForm;
  TempPercent : Integer;

begin
  Frm := TPercentForm.Create(Self);
  Frm.ShowModal;
  if (Frm.ModalResult = mrOK) then
    TempPercent := StrToInt(Frm.PercentEdit.Text)
  else
    TempPercent := -1;
  Frm.Free;

  if (TempPercent <> -1) then
    UpdateViewPercent(TempPercent);
end;

procedure TMainForm.WhitespaceCompOptionClick(Sender: TObject);
var
  Frm : TWhitespaceCompForm;
  Tmp : Integer;

begin
  Frm := TWhitespaceCompForm.Create(Self);
  Frm.CompEnabledBox.Checked := FaxViewer.WhitespaceCompression;
  if FaxViewer.WhitespaceCompression then begin
    Frm.FromEdit.Text := IntToStr(FaxViewer.WhitespaceFrom);
    Frm.ToEdit.Text   := IntToStr(FaxViewer.WhitespaceTo);
  end;

  if (Frm.ShowModal = mrOK) then begin
    FaxViewer.WhitespaceCompression := Frm.CompEnabledBox.Checked;
    if FaxViewer.WhitespaceCompression then begin
      Tmp := StrToInt(Frm.FromEdit.Text);
      FaxViewer.WhitespaceFrom := Tmp;
      Tmp := StrToInt(Frm.ToEdit.Text);
      FaxViewer.WhitespaceTo := Tmp;
    end;
  end;

  Frm.Free;
end;

procedure TMainForm.FormCreate(Sender: TObject);
begin
  ViewPercent := 100;
  DisableZoomChoices;
  SelectAllItem.Enabled := False;
  CopyItem.Enabled      := False;
end;

procedure TMainForm.UpdateViewPercent(const NewPercent : Integer);
begin
  if (NewPercent = ViewPercent) then
    Exit;

  ViewPercent := NewPercent;

  if (NewPercent = 100) then
    FaxViewer.Scaling := False
  else begin
    FaxViewer.BeginUpdate;
    FaxViewer.Scaling   := True;
    FaxViewer.HorizMult := NewPercent;
    FaxViewer.HorizDiv  := 100;
    FaxViewer.VertMult  := NewPercent;
    FaxViewer.VertDiv   := 100;
    FaxViewer.EndUpdate;
  end;

  UncheckZoomChoices;
  case ViewPercent of
    25 : N25PercentItem.Checked  := True;
    50 : N50PercentItem.Checked  := True;
    75 : N75PercentItem.Checked  := True;
    100: N100PercentItem.Checked := True;
    200: N200PercentItem.Checked := True;
    400: N400PercentItem.Checked := True;
    else
      OtherSizeItem.Checked := True;
  end;
end;

procedure TMainForm.EnableZoomChoices;
begin
  N25PercentItem.Enabled  := True;
  N50PercentItem.Enabled  := True;
  N75PercentItem.Enabled  := True;
  N100PercentItem.Enabled := True;
  N200PercentItem.Enabled := True;
  N400PercentItem.Enabled := True;
  OtherSizeItem.Enabled   := True;
  ZoomInItem.Enabled      := True;
  ZoomOutItem.Enabled     := True;
end;

procedure TMainForm.DisableZoomChoices;
begin
  N25PercentItem.Enabled  := False;
  N50PercentItem.Enabled  := False;
  N75PercentItem.Enabled  := False;
  N100PercentItem.Enabled := False;
  N200PercentItem.Enabled := False;
  N400PercentItem.Enabled := False;
  OtherSizeItem.Enabled   := False;
  ZoomInItem.Enabled      := False;
  ZoomOutItem.Enabled     := False;
end;

procedure TMainForm.UncheckZoomChoices;
begin
  N25PercentItem.Checked  := False;
  N50PercentItem.Checked  := False;
  N75PercentItem.Checked  := False;
  N100PercentItem.Checked := False;
  N200PercentItem.Checked := False;
  N400PercentItem.Checked := False;
  OtherSizeItem.Checked   := False;
end;

procedure TMainForm.EnableRotationChoices;
begin
  NoRotateItem.Enabled  := True;
  Rotate90Item.Enabled  := True;
  Rotate180Item.Enabled := True;
  Rotate270Item.Enabled := True;
end;

procedure TMainForm.DisableRotationChoices;
begin
  NoRotateItem.Enabled  := False;
  Rotate90Item.Enabled  := False;
  Rotate180Item.Enabled := False;
  Rotate270Item.Enabled := False;
end;

procedure TMainForm.UncheckRotationChoices;
begin
  NoRotateItem.Checked  := False;
  Rotate90Item.Checked  := False;
  Rotate180Item.Checked := False;
  Rotate270Item.Checked := False;
end;

procedure TMainForm.NoRotateItemClick(Sender: TObject);
begin
  FaxViewer.Rotation := vr0;
  UncheckRotationChoices;
  NoRotateItem.Checked := True;
end;

procedure TMainForm.Rotate90ItemClick(Sender: TObject);
begin
  FaxViewer.Rotation := vr90;
  UncheckRotationChoices;
  Rotate90Item.Checked := True;
end;

procedure TMainForm.Rotate180ItemClick(Sender: TObject);
begin
  FaxViewer.Rotation := vr180;
  UncheckRotationChoices;
  Rotate180Item.Checked := True;
end;

procedure TMainForm.Rotate270ItemClick(Sender: TObject);
begin
  FaxViewer.Rotation := vr270;
  UncheckRotationChoices;
  Rotate270Item.Checked := True;
end;

procedure TMainForm.FaxViewerPageChange(Sender: TObject);
var
  W : Word;
begin
  if (FaxViewer.FileName <> '') then
    begin
      StatusPanel.Caption := Format('  Viewing page %d of %d in %s', [FaxViewer.ActivePage,
        FaxViewer.NumPages, FaxViewer.FileName]);
      W := FaxViewer.PageFlags;
      CheckBox1.Checked := W and ffHighRes <> 0;
      CheckBox2.Checked := W and ffHighWidth <> 0;
      CheckBox3.Checked := W and ffLengthWords <> 0;
    end
  else
    begin
      StatusPanel.Caption := '  No file loaded';
      Panel1.Caption := '';
    end;
end;

procedure TMainForm.FaxViewerViewerError(Sender: TObject;
  ErrorCode: Integer);
begin
  MessageDlg(Format('Viewer error %d', [ErrorCode]), mtError, [mbOK], 0);
end;

procedure TMainForm.NextClick(Sender: TObject);
begin
  if FaxViewer.ActivePage < FaxViewer.NumPages then
    FaxViewer.ActivePage := FaxViewer.ActivePage + 1;
end;

procedure TMainForm.PrevClick(Sender: TObject);
begin
  if FaxViewer.ActivePage > 1 then
    FaxViewer.ActivePage := FaxViewer.ActivePage - 1;
end;

procedure TMainForm.FormShow(Sender: TObject);
begin
  if ParamCount > 0 then
    OpenFile(ParamStr(1));
end;

procedure TMainForm.FaxViewerDropFile(Sender: TObject; FileName: String);
begin
  EnableZoomChoices;
  EnableRotationChoices;
  SelectAllItem.Enabled   := True;
  CopyItem.Enabled        := True;
  N100PercentItem.Checked := True;
  NoRotateItem.Checked    := True;
  ViewPercent             := 100;
  StatusPanel.Caption     := Format('  Viewing page 1 of %d in %s', [FaxViewer.NumPages, FaxViewer.FileName]);
end;

end.

File: examples/cbuilder/VIEWMAIN.cpp
// ***** BEGIN LICENSE BLOCK *****
// * Version: MPL 1.1
// *
// * The contents of this file are subject to the Mozilla Public License Version
// * 1.1 (the "License"); you may not use this file except in compliance with
// * the License. You may obtain a copy of the License at
// * http://www.mozilla.org/MPL/
// *
// * Software distributed under the License is distributed on an "AS IS" basis,
// * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
// * for the specific language governing rights and limitations under the
// * License.
// *
// * The Original Code is TurboPower Async Professional
// *
// * The Initial Developer of the Original Code is
// * TurboPower Software
// *
// * Portions created by the Initial Developer are Copyright (C) 1991-2002
// * the Initial Developer. All Rights Reserved.
// *
// * Contributor(s):
// *
// * ***** END LICENSE BLOCK *****

/*********************************************************/
/*                      VIEWMAIN.CPP                     */
/*********************************************************/
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include <stdio.h>

#include "ViewMain.h"
#include "Percent.h"
#include "WComp.h"
//---------------------------------------------------------------------------
#pragma link "AdFView"
#pragma link "AdFaxPrn"
#pragma link "AdFPStat"
#pragma link "OoMisc"
#pragma resource "*.dfm"
TMainForm *MainForm;
//---------------------------------------------------------------------------
__fastcall TMainForm::TMainForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void TMainForm::UpdateViewPercent(const int NewPercent)
{
  if (NewPercent == ViewPercent)
    return;

  ViewPercent = NewPercent;

  if (NewPercent == 100)
    FaxViewer->Scaling = false;
  else {
    FaxViewer->BeginUpdate();
    FaxViewer->Scaling   = true;
    FaxViewer->HorizMult = NewPercent;
    FaxViewer->HorizDiv  = 100;
    FaxViewer->VertMult  = NewPercent;
    FaxViewer->VertDiv   = 100;
    FaxViewer->EndUpdate();
  }

  UncheckZoomChoices();
  switch (ViewPercent) {
    case 25 : N25PercentItem->Checked  = true; break;
    case 50 : N50PercentItem->Checked  = true; break;
    case 75 : N75PercentItem->Checked  = true; break;
    case 100: N100PercentItem->Checked = true; break;
    case 200: N200PercentItem->Checked = true; break;
    case 400: N400PercentItem->Checked = true; break;
    default :
      OtherSizeItem->Checked = true;
  }
}

void TMainForm::EnableZoomChoices()
{
  N25PercentItem->Enabled  = true;
  N50PercentItem->Enabled  = true;
  N75PercentItem->Enabled  = true;
  N100PercentItem->Enabled = true;
  N200PercentItem->Enabled = true;
  N400PercentItem->Enabled = true;
  OtherSizeItem->Enabled   = true;
  ZoomInItem->Enabled      = true;
  ZoomOutItem->Enabled     = true;
}

void TMainForm::DisableZoomChoices()
{
  N25PercentItem->Enabled  = false;
  N50PercentItem->Enabled  = false;
  N75PercentItem->Enabled  = false;
  N100PercentItem->Enabled = false;
  N200PercentItem->Enabled = false;
  N400PercentItem->Enabled = false;
  OtherSizeItem->Enabled   = false;
  ZoomInItem->Enabled      = false;
  ZoomOutItem->Enabled     = false;
}

void TMainForm::UncheckZoomChoices()
{
  N25PercentItem->Checked  = false;
  N50PercentItem->Checked  = false;
  N75PercentItem->Checked  = false;
  N100PercentItem->Checked = false;
  N200PercentItem->Checked = false;
  N400PercentItem->Checked = false;
  OtherSizeItem->Checked   = false;
}

void TMainForm::EnableRotationChoices()
{
  NoRotateItem->Enabled  = true;
  Rotate90Item->Enabled  = true;
  Rotate180Item->Enabled = true;
  Rotate270Item->Enabled = true;
}

void TMainForm::DisableRotationChoices()
{
  NoRotateItem->Enabled  = false;
  Rotate90Item->Enabled  = false;
  Rotate180Item->Enabled = false;
  Rotate270Item->Enabled = false;
}

void TMainForm::UncheckRotationChoices()
{
  NoRotateItem->Checked  = false;
  Rotate90Item->Checked  = false;
  Rotate180Item->Checked = false;
  Rotate270Item->Checked = false;
}

void __fastcall TMainForm::OpenItemClick(TObject *Sender)
{
  if (OpenDialog->Execute()) {
    OpenFile(OpenDialog->FileName);
  }
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::PrintSetupItemClick(TObject *Sender)
{
  FaxPrinter->PrintSetup();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::PrintItemClick(TObject *Sender)
{
  if (FaxViewer->FileName != "") {
    FaxPrinter->FileName = FaxViewer->FileName;
    FaxPrinter->PrintFax();
  }
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::ExitItemClick(TObject *Sender)
{
	Close();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::ZoomInItemClick(TObject *Sender)
{
  int TempPercent;

  if ((ViewPercent % 25) == 0)
    TempPercent = ViewPercent + 25;
  else
    TempPercent = ViewPercent + (25 - (ViewPercent % 25));
  if (TempPercent > 400) {
    MessageBeep(0);
    return;
  }
  UpdateViewPercent(TempPercent);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::ZoomOutItemClick(TObject *Sender)
{
  int TempPercent;

  if ((ViewPercent % 25) == 0)
    TempPercent = ViewPercent - 25;
  else
    TempPercent = ViewPercent - (25 - (ViewPercent % 25));

  if (TempPercent < 25) {
    MessageBeep(0);
    return;
  }
  UpdateViewPercent(TempPercent);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N25PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(25);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N50PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(50);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N75PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(75);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N100PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(100);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N200PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(200);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::N400PercentItemClick(TObject *Sender)
{
  UpdateViewPercent(400);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::OtherSizeItemClick(TObject *Sender)
{
  TPercentForm* Frm = new TPercentForm(this);
  int TempPercent;

  Frm->ShowModal();
  if (Frm->ModalResult == mrOk)
    TempPercent = Frm->PercentEdit->Text.ToInt();
  else
    TempPercent = -1;
  delete Frm;

  if (TempPercent != -1) 
    UpdateViewPercent(TempPercent);
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::WhitespaceCompOptionClick(TObject *Sender)
{
  TWhitespaceCompForm* Frm = new TWhitespaceCompForm(this);
  int Tmp;

  Frm->CompEnabledBox->Checked = FaxViewer->WhitespaceCompression;
  if (FaxViewer->WhitespaceCompression) {
    Frm->FromEdit->Text = String((int)FaxViewer->WhitespaceFrom);
    Frm->ToEdit->Text   = String((int)FaxViewer->WhitespaceTo);
  }

  if (Frm->ShowModal() == mrOk) {
    FaxViewer->WhitespaceCompression = Frm->CompEnabledBox->Checked;
    if (FaxViewer->WhitespaceCompression) {
      Tmp = Frm->FromEdit->Text.ToInt();
      FaxViewer->WhitespaceFrom = Tmp;
      Tmp = Frm->ToEdit->Text.ToInt();
      FaxViewer->WhitespaceTo = Tmp;
    }
  }
  delete Frm;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::SelectAllItemClick(TObject *Sender)
{
  FaxViewer->SelectImage();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::CopyItemClick(TObject *Sender)
{
  FaxViewer->CopyToClipBoard();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::NoRotateItemClick(TObject *Sender)
{
  FaxViewer->Rotation = vr0;
  UncheckRotationChoices();
  NoRotateItem->Checked = true;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::Rotate90ItemClick(TObject *Sender)
{
  FaxViewer->Rotation = vr90;
  UncheckRotationChoices();
  Rotate90Item->Checked = true;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::Rotate180ItemClick(TObject *Sender)
{
  FaxViewer->Rotation = vr180;
  UncheckRotationChoices();
  Rotate180Item->Checked = true;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::Rotate270ItemClick(TObject *Sender)
{
  FaxViewer->Rotation = vr270;
  UncheckRotationChoices();
  Rotate270Item->Checked = true;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FormCreate(TObject *Sender)
{
  NextPage->Enabled = false;
  PrevPage->Enabled = false;
  ViewPercent = 100;
  DisableZoomChoices();
  SelectAllItem->Enabled = false;
  CopyItem->Enabled = false;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FaxViewerPageChange(TObject *Sender)
{
  if (FaxViewer->FileName != "") {
    char buff[MAX_PATH + 30];
    sprintf(buff, "  Viewing page %d of %d in %s",
      FaxViewer->ActivePage, FaxViewer->NumPages, FaxViewer->FileName.c_str());
    StatusBar->SimpleText    = buff;
  }
  else
    StatusBar->SimpleText = "  No file loaded";
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FaxViewerViewerError(TObject *Sender, int ErrorCode)
{
  MessageDlg("Viewer Error", mtError, TMsgDlgButtons() << mbOK, 0);
}
//---------------------------------------------------------------------------
void TMainForm::OpenFile(String FileName)
{
  FaxViewer->BeginUpdate();
  FaxViewer->Scaling   = false;
  FaxViewer->HorizMult = 1;
  FaxViewer->HorizDiv  = 1;
  FaxViewer->VertMult  = 1;
  FaxViewer->VertDiv   = 1;
  FaxViewer->EndUpdate();
  UncheckZoomChoices();
  UncheckRotationChoices();
  try {
    FaxViewer->FileName = OpenDialog->FileName;
    EnableZoomChoices();
    EnableRotationChoices();
    SelectAllItem->Enabled   = true;
    CopyItem->Enabled        = true;
    N100PercentItem->Checked = true;
    NoRotateItem->Checked    = true;
    ViewPercent              = 100;
    char buff[MAX_PATH + 30];
    sprintf(buff, "  Viewing page 1 of %d in %s",
      FaxViewer->NumPages, FaxViewer->FileName.c_str());
    StatusBar->SimpleText    = buff;
    NextPage->Enabled = true;
    PrevPage->Enabled = true;
  }
  catch (...) {
    CloseFile();
    MessageDlg("Error opening fax file",
      mtError, TMsgDlgButtons() << mbOK, 0);
  }
}

void TMainForm::CloseFile()
{
  FaxViewer->FileName = "";
  DisableZoomChoices();
  DisableRotationChoices();
  SelectAllItem->Enabled = false;
  CopyItem->Enabled      = false;
  StatusBar->SimpleText = "  No file loaded";
  FaxViewer->Invalidate();
  NextPage->Enabled = false;
  PrevPage->Enabled = false;
}

void __fastcall TMainForm::CloseItemClick(TObject *Sender)
{
  CloseFile();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::NextPageClick(TObject *Sender)
{
  if (FaxViewer->ActivePage < FaxViewer->NumPages)
    FaxViewer->ActivePage++;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::PrevPageClick(TObject *Sender)
{
  if (FaxViewer->ActivePage > 1)
    FaxViewer->ActivePage--;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FaxViewerDropFile(TObject *Sender,
  AnsiString FileName)
{
  EnableZoomChoices();
  EnableRotationChoices();
  SelectAllItem->Enabled   = true;
  CopyItem->Enabled        = true;
  N100PercentItem->Checked = true;
  NoRotateItem->Checked    = true;
  ViewPercent              = 100;
  char buff[MAX_PATH + 30];
  sprintf(buff, "  Viewing page 1 of %d in %s",
    FaxViewer->NumPages, FaxViewer->FileName.c_str());
  StatusBar->SimpleText    = buff;
  NextPage->Enabled = true;
  PrevPage->Enabled = true;
}
//---------------------------------------------------------------------------
